File: Raven.Abstractions/Data/ScriptedPatchRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Raven.Json.Linq;
using Raven.Abstractions.Extensions;

namespace Raven.Abstractions.Data
{
	/// <summary>
	/// A Advanced patch request for a specified document (using JavaScript)
	/// </summary>
	public class ScriptedPatchRequest
	{
		/// <summary>
		/// The JavaScript function to use the patch a document
		/// </summary>
		/// <value>The type.</value>
		public string Script { get; set; }

		public Dictionary<string, object> Values { get; set; }

		public ScriptedPatchRequest()
		{
			Values = new Dictionary<string, object>();
		}

		public static ScriptedPatchRequest FromJson(RavenJObject patchRequestJson)
		{
			return patchRequestJson.JsonDeserialization<ScriptedPatchRequest>();
		}

		protected bool Equals(ScriptedPatchRequest other)
		{
			if(other == null)
				return false;
			return string.Equals(Script, other.Script) && Values.Keys.SequenceEqual(other.Values.Keys);
		}

		public override bool Equals(object obj)
		{
			if (ReferenceEquals(null, obj)) return false;
			if (ReferenceEquals(this, obj)) return true;
			return Equals(obj as ScriptedPatchRequest);
		}

		public override int GetHashCode()
		{
			unchecked
			{
				return Values.Keys.Aggregate(Script.GetHashCode()*397, (i, s) => i*397 ^ s.GetHashCode());
			}
		}
	}
}

File: Raven.Client.Java/lib/src/main/java/raven/abstractions/data/ScriptedPatchRequest.java
package raven.abstractions.data;

import java.io.IOException;
import java.util.HashMap;
import java.util.Map;

import org.apache.commons.lang.builder.EqualsBuilder;
import org.apache.commons.lang.builder.HashCodeBuilder;

import raven.abstractions.extensions.JsonExtensions;
import raven.abstractions.json.linq.RavenJObject;

/**
 * A Advanced patch request for a specified document (using JavaScript)
 */
public class ScriptedPatchRequest {

  private String script;
  private Map<String, Object> values;

  public ScriptedPatchRequest() {
    values = new HashMap<String, Object>();
  }
  public String getScript() {
    return script;
  }
  public Map<String, Object> getValues() {
    return values;
  }
  public void setScript(String script) {
    this.script = script;
  }
  public void setValues(Map<String, Object> values) {
    this.values = values;
  }

  public static ScriptedPatchRequest fromJson(RavenJObject patchRequestJson) {
    try {
      return JsonExtensions.getDefaultObjectMapper().readValue(patchRequestJson.toString(), ScriptedPatchRequest.class);
    } catch (IOException e ){
      throw new RuntimeException("Unable to parse ScriptedPatchRequest", e);
    }
  }

  @Override
  public int hashCode() {
    return new HashCodeBuilder().append(script).append(values).hashCode();
  }
  @Override
  public boolean equals(Object obj) {
    if (this == obj)
      return true;
    if (obj == null)
      return false;
    if (getClass() != obj.getClass())
      return false;
    ScriptedPatchRequest other = (ScriptedPatchRequest) obj;
    return new EqualsBuilder().append(script, other.script).append(values, other.values).isEquals();
  }

}
